### ISSUE TITLE
Replace $VERSION by ${project.version} in WEB-INF/web.xml

### ISSUE THREAD

**Issue by thadguidry** [OpenRefine/OpenRefine#6878]
##Context
Eventually, we'd like contributors to actually be able to run `mvn package` on Windows, instead of currently only Mac users.  This issue was opened to begin to investigate the `/packaging/pom.xml` and start to untangle things better for a profile for Windows, which currently fails to successfully produce an OpenRefine Windows package on Windows (sigh).  But first, we should take care of TODO's perhaps?  Previous maintainers left some TODO's without creating an issue to resolve it.  So this is that issue and my attempt to research and find a resolution.

We currently have a workaround for the `web.xml` version string replacement during packaging here:
https://github.com/OpenRefine/OpenRefine/blob/11168a425bfc829a8d81bc58c2cacb93cca19b52/packaging/pom.xml#L232-L23

I did some research, and it seems the Maven War Plugin could be used to automatically replace property placeholders (string interpolation) within the `web.xml` file?

Current TODO's are found under
- https://github.com/OpenRefine/OpenRefine/blob/11168a425bfc829a8d81bc58c2cacb93cca19b52/packaging/linux.xml#L66
- https://github.com/OpenRefine/OpenRefine/blob/11168a425bfc829a8d81bc58c2cacb93cca19b52/packaging/windows-with-jre.xml#L79

Example snippets that could be added to our main pom.xml :

### Example 1
```
      <!-- War plugin used to automatically replace property placeholders,
       such as ${project.version} within /WEB-INF when running 'mvn package'
       -->
      <plugin>
        <artifactId>maven-war-plugin</artifactId>
        <version>3.4</version>
        <configuration>
        <!-- this is weird but "filtering" here actually also does "replacing" of property placeholders -->
          <filteringDeploymentDescriptors>true</filteringDeploymentDescriptors>
        </configuration>
      </plugin>
```

Or more specifically it seems that we could also just apply "filtering" to resources that we want or need:
### Example 2
```
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-war-plugin</artifactId>
    <configuration>
        <webResources>
            <resource>
                <filtering>true</filtering>
                <directory>src/main/webapp</directory>
                <includes>
                    <include>**/web.xml</include>
                </includes>
            </resource>
        </webResources>
    </configuration>
</plugin>
```

References:
1. https://maven.apache.org/plugins/maven-war-plugin/faq.html
2. https://stackoverflow.com/questions/15355708/how-to-replace-a-value-in-web-xml-with-a-maven-property
3. https://stackoverflow.com/questions/40430305/how-to-do-string-replacements-with-maven-war-plugin-2-6

**Comment by JavierABarreto** [comment]
Hi @thadguidry, I would like to take this issue and contribute!

**Comment by JavierABarreto** [comment]
Hi @thadguidry, do I needed to create a test or just making sure that the other tests are running with green light will solve it?

**Comment by thadguidry** [comment]
- tests should run with a green light.
- It's likely that we instead might want to use `$revision` handling for our CI.  Maven has a guide on this even specifically for Multi Module Setup like what we do https://maven.apache.org/maven-ci-friendly.html#multi-module-setup
- We might need that War plugin, or perhaps not.  These needs investigation by whomever is working on this issue.
- This issue should make things work on Windows, Linux, Mac, for our packages on those platforms.
- We don't need to worry about `mvn deploy` on Windows, only `mvn package` I think?  So that developers on Windows can easily test, create a Windows package with the right versioning and can test the currently configured INNO Setup installer.
- It's probably most helpful to look under our `.github/workflows` to see how we build the packages for the 3 OS's.

**Comment by thadguidry** [comment]
@JavierABarreto If you have the patience (this will likely suck away hours and hours of debugging on Windows with Maven, Java, config, POM's, you-name-it), but we would be **forever indebted** to you  for making this possible.  I'd expect that it might even take weeks of further reading/research, trial-and-error.  GO FOR IT!  We're here to help as needed.

**Comment by JavierABarreto** [comment]
Hi @thadguidry, when I try to run it on windows and build the package, it fails on grey tests, do you happen to know why?

**Comment by thadguidry** [comment]
Can you give me the mvn command you ran? Which IDE and what did you do to launch the mvn command to build?
You can also skip the tests for now.

**Comment by JavierABarreto** [comment]
Sure, the command that I used was "mvn package", I use IntelliJ IDEA CE IDE, and what I did was:
- Clone the repository
- Open it in IntelliJ
- Wait for it to install all the dependencies, plugins
- Set the Java 11 version to use in the project structure
- Run the command `mvn package`

**Comment by thadguidry** [comment]
Ok then it sounds like you have successfully reproduced the issue.

What’s your proposal to fix?

Thad Guidry
________________________________

**Comment by kalindafab** [comment]
Hi @JavierABarreto are you still working on this issue? I’ve done some research and found that I could work on it as well. Let me know if it’s okay for me to take over.

**Comment by kalindafab** [comment]
Hi @thadguidry I’d like to share my proposal for addressing this issue:

Tackling TODOs: I replaced $VERSION with ${project.version} and $REVISION with ${revision} to make the configuration more consistent and flexible.
Maven Filtering Implementation: I added the Maven War Plugin to handle filtering effectively.
I ran the command:
mvn -Drevision=3.9.8 clean package,
and while the build was successful, no JAR or WAR file was created in the target folder. Instead, some other files were generated. I believe further research could help improve this.

Let me know if you’d like me to continue working on this or if you have any guidance on resolving the missing artifact issue.
Thanks!

**Comment by JavierABarreto** [comment]
@kalindafab Yes, of course! Have been busy irl that I couldn't work on this issue... @thadguidry could you assign @kalindafab to this issue and remove me? sorry for the inconvenience

**Comment by thadguidry** [comment]
@kalindafab Hi and thanks for picking up this issue and trying to work on it.
If you'd like some collaboration on this issue, I'll be more than happy to help.
Why not start with opening a Draft PR so that you can share your progress thus far, and I can pull in that PR and also run it and help you debug.  We can then continue discussion in your Draft PR itself.
